src/engine.ts

```diff
@@ -223,12 +238,49 @@ export default class Engine {
     //TODO manage gaia phase actions for specific factions
   }
 
+  leechingPhase(player: PlayerEnum, location: CubeCoordinates){
+    // exclude setiup rounds
+    if (this.round <=0) {
+      return;
+    } 
+    // all players excluded leecher
+    this.roundSubCommands = [];
+
+    for (const pl of this.players){     
```

Comment:
it should be `this.playersInOrder()`

Reply:
actually the rules state each other player with at least, probably meaning no order. Anyway, turn order could be wrong because even players that have passed can leech, and turnOrder is updated on every pass

Reply:
you're right